includes/export/class-wc-product-csv-exporter.php

```diff
@@ -268,7 +268,13 @@ protected function get_column_value_published( $product ) {
 			'publish' => 1,
 		);
 
-		$status = $product->get_status( 'edit' );
+		// Fix display for variations when parent product is a draft.
+		if ( 'variation' === $product->get_type() ) {
```

Comment:
@claudiosanches You could add `'draft' === $parent['status']` to the if condition and simplify the code block.

Reply:
We need to check first if it's a variation because we are using `WC_Product_Variation::get_parent_data`, so moving it from this code block will force us to at least check if the `get_parent_data` is callable, also increasing conditional code.